twingate/internal/client/service-account.go

```diff
@@ -72,6 +72,19 @@ func (client *Client) UpdateServiceAccount(ctx context.Context, serviceAccount *
 }
 
 func (client *Client) DeleteServiceAccount(ctx context.Context, serviceAccountID string) error {
+	serviceAccount, err := client.ReadServiceAccount(ctx, serviceAccountID)
+	if err != nil && !errors.Is(err, ErrGraphqlResultIsEmpty) {
+		return err
+	}
+
+	if serviceAccount != nil {
```

Comment:
I dont understand this fix and the relation to #407.
This code is correct - if we want TF to delete an SA, it has to revoke all its keys first.
But what happened in the issue is something else - the SA was deleted outside of TF:
- In this case `serviceAccount` is `nil` and this block doesnt even run
- `DeleteServiceAccount` mutation will succeed -> we'll delete the SA from TF state **but not** its keys

Reply:
We need a reproduction unit tests that:
1. Creates an SA + key using TF
2. Makes "manual" API calls to revoke the key and delete the SA
3. Checked TF re-creates both SA + key

Reply:
updated PR